packages/rule-stylesheet-limits/src/rule.ts

```diff
@@ -0,0 +1,177 @@
+                maxRules = options.maxRules;
+                hasRuleLimit = true;
+            }
+            if (options.maxSheets && (!hasSheetLimit || options.maxSheets < maxSheets)) {
+                maxSheets = options.maxSheets;
+                hasSheetLimit = true;
+            }
+        }
+
+        // The following function will be evaluated in the context of the page.
+        const injectedCode = function() {
+
+            // Recursively count rules and imports in the passed stylesheet.
+            const countRules = (styleSheet: CSSStyleSheet) => {
+
+                const results = {
+                    imports: 0,
+                    rules: 0,
+                    sheets: 1
+                };
+
+                try {
+
+                    // Count rules in this stylesheet.
+                    Array.from(styleSheet.cssRules).forEach((rule) => {
+
+                        if (rule instanceof CSSStyleRule) {
+
+                            // Count each selector in a style rule separately.
+                            results.rules += rule.selectorText.split(',').length;
+
+                        } else if (rule instanceof CSSImportRule) {
+
+                            // Recursively count rules in imported stylesheets.
+                            const subResults = countRules(rule.styleSheet);
+
+                            results.imports += Math.max(results.imports, subResults.imports + 1);
+                            results.rules += subResults.rules + 1;
+                            results.sheets += subResults.sheets;
+
+                        } else {
+
+                            // Other rules count as one each.
+                            results.rules += 1;
+
+                        }
+                    });
+
+                } catch (e) {
+
+                    /*
+                     * Accessing cssRules of a cross-origin stylesheet can throw.
+                     * If this happens, exclude the stylesheet from the count.
+                     */
+                }
+
+                return results;
+            };
+
+            const combinedResults = {
+                imports: 0,
+                rules: 0,
+                sheets: 0
+            };
+
+            // Get the recursive count of rules for all stylesheets in the page.
+            Array.from(document.styleSheets).forEach((sheet) => {
+                if (sheet instanceof CSSStyleSheet) {
+                    const subResults = countRules(sheet);
+
+                    combinedResults.imports += Math.max(combinedResults.imports, subResults.imports);
+                    combinedResults.rules += subResults.rules;
+                    combinedResults.sheets += subResults.sheets;
+                }
+            });
+
+            return combinedResults;
+        };
+
+        const validateScanEnd = async () => {
+
+            const results = await context.evaluate(`(${injectedCode})()`);
```

Comment:
Ok, I'll take a look. Thanks for figuring this out!

Reply:
Digging in I see that `injectedCode` when run via `yarn test` has additional statements like `cov_l3mx4wetg.s[46]++;` included by the code coverage utility `nyc`. These generate a script error when the code is actually injected since it runs in a different context.

Adding `/* istanbul ignore next */` before the definition of `injectedCode` fixes the issue. I'll update the PR shortly.